Userland/Services/RequestServer/Protocol.cpp

```diff
@@ -38,6 +38,7 @@ ErrorOr<Protocol::Pipe> Protocol::get_pipe_for_request()
         dbgln("Protocol: pipe() failed: {}", strerror(saved_errno));
         return Error::from_errno(saved_errno);
     }
+    fcntl(fd_pair[0], F_SETFL, fcntl(fd_pair[0], F_GETFL) | O_NONBLOCK);
```

Comment:
Drive-by nit: Do we have a Core::System::pipe (or pipe2) that lets us pass O_NONBLOCK at pipe creation time (if supported?)

Reply:
We have `Core::System::pipe2` which...conditionally drops its flags depending on the OS 😑
Lemme see if anything breaks if we make that call `fcntl` for us.

Reply:
Fixed